### ISSUE TITLE
Show overlapping countries in scatterplots

### ISSUE THREAD

**Issue by saloni-nd** [owid/owid-grapher#2203]
## Description

@lucasrodes and I noticed that for scatterplots, when countries have the exact same values on both axes, they can’t be seen and there’s no indication that there are many countries there.

- For example, in this [chart](https://ourworldindata.org/grapher/prevalence-of-schizophrenia-in-males-vs-females) on the prevalence of schizophrenia in men vs. women –– not many countries are visible, because many of them have the same values.
- But in this [chart](https://ourworldindata.org/grapher/prevalence-of-depression-males-vs-females) on the prevalence of depression in men vs. women –– many more countries are visible.

![prevalence-of-schizophrenia-in-males-vs-females (1)](https://github.com/owid/owid-grapher/assets/31684905/923068a9-816a-4eb6-a31b-24903d025307)
![prevalence-of-depression-males-vs-females (1)](https://github.com/owid/owid-grapher/assets/31684905/8d2f0734-692c-4008-ab21-9e3e69c22b73)

## Expected behaviour

It should be made clearer that there are countries overlapping.

A few ideas:
- The opacity could be changed, so that places where many countries are overlapping would be darker
- There could be a jitter added, but that would be misleading since the scale is continuous
- The overlapping countries could be labelled with a line/arrow pointing to them.
- Or something else.

## Environment

Desktop:
iOS, Chrome Version 111.0.5563.146

**Comment by larsyencken** [comment]
After chatting with Sophia, this one seems very tricky, since it's not at all clear what the data viz solution is to plotting overlapping points like this.

@samizdatco We're tempted to say there's not much we can do here. Any magic tooltip suggestions for when data points are overlapping?

**Comment by danyx23** [comment]
Yeah tooltips sounds like a good place to tackle this. It's a bit tricky how to do this well without letting it turn into a major project. Basically what you want is to find the data points that, in 2D space, are closer together than 1 px and then include and list those in the tooltip. This kind of aggregation could be a bit expensive to compute though - you also want a sparse data structure with fast lookups. Maybe iterating over the data points, getting their X/Y coordinates and saving that in a map where the key is a stringified version of the rounded "x-y" coordinate pair would work well enough? This would have to be a computed property for when the dots change but could then be efficiently looked up for the tooltip
